tests/test_download_client.py

```diff
@@ -143,12 +142,47 @@ def test_download_tarfile(self):
             for member in t.getmembers():
                 m = t.extractfile(member)
                 contents = m.read()
-                if sys.version_info[0] < 3:
-                    assert contents == uuids[m.name]['contents']
-                else:
-                    assert contents.decode('utf-8') == uuids[member.name]['contents']
+                assert contents.decode('utf-8') == uuids[member.name]['contents']
                 os.remove(tarfile_name)
 
+    def test_download_annotations(self):
+
+        # uuid of file that has an annotation
+        files_to_dl = ['small_ann']
+
+        # get annotation id out of metadata
+        index_client = GDCIndexClient(base_url)
+        index_client._get_metadata(files_to_dl)
+
+        # set expected output path for download client
+        override_kwargs = copy.deepcopy(client_kwargs)
+        override_kwargs['directory'] = os.getcwd() + '/tests'
+        # where we expect annotations to be written
+        output_path = override_kwargs['directory'] + '/{}/annotations.txt'.format(
```

Comment:
you should probably use pytest's tempfil/tmpdir fixtures for this, instead of creating an empty directory for tests to use. tests can fail for various reasons and you might not get to the cleanup section at all. as an alternative you can also do something around python's `tempfile` library (python3 has a context manager support for a directory)